clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.block;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.client.BlockMasterClient;
+import tachyon.client.ClientContext;
+import tachyon.client.ClientOptions;
+import tachyon.thrift.BlockInfo;
+import tachyon.thrift.NetAddress;
+import tachyon.util.network.NetworkAddressUtils;
+
+/**
+ * Tachyon Block Store client. This is an internal client for all block level operations in Tachyon.
+ * An instance of this class can be obtained via {@link TachyonBlockStore#get}. The methods in this
+ * class are completely opaque to user input (such as {@link ClientOptions}). This class is thread
+ * safe.
+ */
+public class TachyonBlockStore implements Closeable {
+
+  private static TachyonBlockStore sCachedClient = null;
+
+  /**
+   * @return a new instance of Tachyon block store
+   */
+  public static synchronized TachyonBlockStore get() {
+    if (null == sCachedClient) {
+      sCachedClient = new TachyonBlockStore();
+    }
+    return sCachedClient;
+  }
+
+  private final BSContext mContext;
```

Comment:
Since `BSContext.INSTANCE` is a singleton, `mContext` is not really a part of the state of this class. So why bother to have such a member variable? Right now it serves more like a short-hand for `BSContext.INSTANCE`